dumb-jump.el

```diff
@@ -125,25 +125,25 @@ or most optimal searcher."
   :type 'string)
 
 (defcustom dumb-jump-ag-word-boundary
-  "(?![\\w\\?\\*-])"
+  "(?![a-zA-Z0-9\\?\\*-])"
```

Comment:
The motivation for using `a-zA-Z0-9` instead of `\w` is that you don't want it to match for `_` which is the only other character not matched by this change?

Reply:
For _some_ searchers it was causing bad matches as outlined in #186 . For example, saying `tester` was a match when jumping on `test`. You can see evidence of this in the early commits for this PR with the :x:s, but here's a deep link to one example:  https://travis-ci.org/jacktasia/dumb-jump/jobs/316563561 
Thanks again for looking at this!

Reply:
This leads me to believe for these searchers with the options we give them `\w` is not recognized at all.

Reply:
Very weird but I see the point. Np :)